### ISSUE TITLE
Support for Consistency Large Language Models?

### ISSUE THREAD

**Issue by unoexperto** [ggerganov/llama.cpp#7168]
It's beyond my level of understanding of DL but have you guys heard about CLLM ? Author(s) claim 3.5X speedup of inference without changing the architecture.

Links:
https://www.reddit.com/r/MachineLearning/comments/1cnfmec/research_consistency_llms_converting_llms_to/
https://hao-ai-lab.github.io/blogs/cllm/

**Comment by JohannesGaessler** [comment]
A 2.4-3.4x speedup, while impressive, cannot be judged in a vacuum. You always need to consider the tradeoffs between speed, quality, and memory use. The difference in memory use should be negligible. However, since you need to specifically finetune the model in order to enable this speedup this degrades the quality of the outputs. The paper reports an increase in perplexity from 8.0 to 9.5. These values are not directly comparable to llama.cpp values since they depend on a lot of factors but I think that this is **a lot** more than I would intuitively expect from e.g. quantization.

My personal philosophy is that I want to enable the evaluation of large models with negligible precision loss compared to the original weights at the lowest possible cost. So I personally am not interested in implementing something like this.

**Comment by sdmorrey** [comment]
My read is the change is to training not code. So it's a fine tuning procedure on the model itself? Is that right or am I missing something?

**Comment by JohannesGaessler** [comment]
llama.cpp would still need code changes specifically to support the technique described in the paper.

**Comment by sdmorrey** [comment]
I'll poke my head over there and see if they're willing to help with that because I'd be very interested in running this. Even with somewhat lower accuracy because I get billed by the millisecond for inference so anything that can speed it up would be a huge savings.

**Comment by JohannesGaessler** [comment]
To clarify my position: I am not willing to do an implementation myself but I would be willing to review someone else's implementation.
